src/core/service.c

```diff
@@ -2693,6 +2697,16 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
                                         else
                                                 service_enter_signal(s, SERVICE_FINAL_SIGTERM, f);
                                         break;
+                                } else if (s->type == SERVICE_NOTIFY) {
+                                        /* Only enter running through a notification, so that the
+                                         * SERVICE_START state signifies that no ready notification
+                                         * has been received */
+                                        if (f != SERVICE_SUCCESS)
+                                                service_enter_signal(s, SERVICE_FINAL_SIGTERM, f);
+                                        else if (!s->remain_after_exit)
+                                                /* The service has never been active */
+                                                service_enter_signal(s, SERVICE_FINAL_SIGTERM, SERVICE_FAILURE_RESOURCES);
+                                        break;
```

Comment:
One more observation: if we want READY=1 to cause execution of any ExecStartPost= commands, then by [src/core/service.c:3066](https://github.com/joukewitteveen/systemd/blob/28ab8adfb4c28482320b6fe5d12f058a145e7341/src/core/service.c#L3066) we should keep the state at SERVICE_START. In other words: the currently implemented logic is already to use SERVICE_START as an indication of the fact that no READY=1 notification has been received.